model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/userSession/HotRodUserSessionTransaction.java

```diff
@@ -79,20 +77,33 @@ private MapUserSessionEntity wrapUserSessionEntityToClientSessionAwareDelegate(M
         if (entity == null) return null;
 
         return new MapUserSessionEntityDelegate(new SimpleDelegateProvider<>(entity)) {
+
+            private boolean filterAndRemoveNotExpired(MapAuthenticatedClientSessionEntity clientSession) {
+                if (!clientSessionTransaction.exists(clientSession.getId())) {
```

Comment:
I noticed that `exists` method now defaults to "read by id". Shouldn't we override `exists` method in `NoActionHotRodTransactionWrapper`?

Reply:
we should, I created a new issue for that: https://github.com/keycloak/keycloak/issues/19196